### DISCUSSION TITLE
Selectively disable distributables produced by `make site`

### DISCUSSION THREAD

**Discussion by eskatos** [hydraulic-software/conveyor#134] [Q&A]
I would like to disable producing the TAR.GZ for Linux and only keep the DEB.
I would like to disable producing the ZIP for Windows and only keep the installers & MSIX.

Is this supported? I did not find anything in the documentation
If not, could it be considered?

**Comment by mikehearn** [comment]
It's not supported at the moment. It could, yes. When you say disable, do you mean you're using the default HTML template and would like to remove the options? Otherwise you can just not link to them. The ZIP and tarball aren't particularly slow to build.

**Reply by eskatos** [reply]
Yes I'm using the default HTML template for now.

I would like to disable building the archives completely. Every bit adds to the build time. Moreover it also adds to upload time when syncing the update site.

**Reply by mikehearn** [reply]
OK, we can add that. Just be aware that if you don't have the .tar.gz version, Linux users who are able to use your app currently will find it much harder to do so. The binaries work on more than just Debian/Ubuntu, usually.

**Reply by mikehearn** [reply]
BTW, may I enquire as to the underlying driver of this? Is the need to optimize build time coming from reducing the cost of per-second charged CI services, or because you're doing a deploy on every commit? Maybe there are other opportunities for optimization here too. We've thought about running a service version of Conveyor where we'd probably be able to do builds much faster than other CI services can, for example, and would charge per build rather than per second.

**Reply by eskatos** [reply]
I do some end to end testing on every PR push against self-signed outputs and staging deployment on every PR merge.
Disabling some outputs could make CI a bit faster. It's not only about cost though, it's also about the feedback loop duration. I'm happy about the current service-less model of Conveyor.

For production, having seamless updates is a must. For Windows production I simply don't need the ZIP. For Linux production I understand that users of non-DEB distros will have basically no install option. I think I can live with that for a while and wait for actual user feedback before considering supporting the tarball and implement freedesktop integration, updates control etc...

Note that from my point of view, this is in no way a deal breaker for using Conveyor, just an idea of an optimisation.
